src/DeviceListener.js

```diff
@@ -197,32 +232,65 @@ export default class DeviceListener {
             }
         }
 
+        // Unverified devices that were there last time the app ran
+        // (technically could just be a boolean: we don't actually
+        // need to remember the device IDs, but for the sake of
+        // symmetry...).
+        const oldUnverifiedDeviceIds = new Set();
+        // Unverified devices that have appeared since then
+        const newUnverifiedDeviceIds = new Set();
+
         // as long as cross-signing isn't ready,
         // you can't see or dismiss any device toasts
         if (crossSigningReady) {
-            let haveUnverifiedDevices = false;
-
-            const devices = await cli.getStoredDevicesForUser(cli.getUserId());
+            const devices = cli.getStoredDevicesForUser(cli.getUserId());
             for (const device of devices) {
                 if (device.deviceId == cli.deviceId) continue;
 
                 const deviceTrust = await cli.checkDeviceTrust(cli.getUserId(), device.deviceId);
                 if (!deviceTrust.isCrossSigningVerified() && !this._dismissed.has(device.deviceId)) {
-                    haveUnverifiedDevices = true;
-                    break;
+                    if (this._ourDeviceIdsAtStart.has(device.deviceId)) {
+                        oldUnverifiedDeviceIds.add(device.deviceId);
+                    } else {
+                        newUnverifiedDeviceIds.add(device.deviceId);
+                    }
                 }
             }
+        }
 
-            if (haveUnverifiedDevices) {
-                ToastStore.sharedInstance().addOrReplaceToast({
-                    key: OTHER_DEVICES_TOAST_KEY,
-                    title: _t("Review where you’re logged in"),
-                    icon: "verification_warning",
-                    component: sdk.getComponent("toasts.UnverifiedSessionToast"),
-                });
-            } else {
-                ToastStore.sharedInstance().dismissToast(OTHER_DEVICES_TOAST_KEY);
+        // Display or hide the batch toast for old unverified sessions
+        if (oldUnverifiedDeviceIds.size > 0) {
+            ToastStore.sharedInstance().addOrReplaceToast({
+                key: OTHER_DEVICES_TOAST_KEY,
+                title: _t("Review where you’re logged in"),
+                icon: "verification_warning",
+                props: {
+                    deviceIds: oldUnverifiedDeviceIds,
+                },
+                component: sdk.getComponent("toasts.BulkUnverifiedSessionsToast"),
+            });
+        } else {
+            ToastStore.sharedInstance().dismissToast(OTHER_DEVICES_TOAST_KEY);
+        }
+
+        // Show toasts for new unverified devices if they aren't already there
+        for (const deviceId of newUnverifiedDeviceIds) {
+            ToastStore.sharedInstance().addOrReplaceToast({
+                key: toastKey(deviceId),
+                title: _t("Unverified login. Was this you?"),
```

Comment:
[Figma](https://www.figma.com/file/wwqBo5oAFfk8XKmtd2YaIs/Cross-signing?node-id=4592%3A104) says "New login" here.

```suggestion
                title: _t("New login. Was this you?"),
```